src/dawn/IIR/ControlFlowDescriptor.cpp

```diff
@@ -19,15 +19,15 @@ namespace iir {
 
 ControlFlowDescriptor ControlFlowDescriptor::clone() const {
   ControlFlowDescriptor copy;
-  for(const auto stmt : controlFlowStatements_) {
+  for(const auto& stmt : controlFlowStatements_) {
     copy.controlFlowStatements_.push_back(stmt->clone());
   }
   return copy;
 }
 void ControlFlowDescriptor::removeStencilCalls(const std::set<int>& stencilIDs,
                                                iir::StencilMetaInformation& metadata) {
-  for(auto it = getStatements().begin(); it != getStatements().end(); ++it) {
-    std::shared_ptr<iir::Stmt> stmt = (*it)->ASTStmt;
+  for(StmtConstIterator it = getStatements().begin(); it != getStatements().end(); ++it) {
+    const std::shared_ptr<iir::Stmt>& stmt = *it;
```

Comment:
2 problems:
- I am forced to have a const ptr here because it comes from dereferencing `it` which has to be a `const_iterator` (needed by call to `eraseStmt(it)` below)
- As I am not changing the stmt it would also make sense to have `std::shared_ptr<const iir::Stmt>` but everywhere in the code we use the non-const version. So, in this case, I could not call `getStencilIDFromStencilCallStmt(callDecl)` below

Reply:
Ok, just didn't see the reason for changing.